pkg/sharing/upload.go

```diff
@@ -210,42 +209,27 @@ func (s *Sharing) uploadFile(inst *instance.Instance, m *Member, file map[string
 			"Authorization": "Bearer " + creds.AccessToken.AccessToken,
 		},
 		Body: bytes.NewReader(body),
-	})
+	}
+	var res *http.Response
+	res, err = request.Req(opts)
 	if err != nil {
 		return err
 	}
+	if res.StatusCode/100 == 5 {
+		return ErrInternalServerError
+	}
 	if res.StatusCode/100 == 4 {
 		res.Body.Close()
-		if err = creds.Refresh(inst, s, m); err != nil {
-			return err
-		}
-		res, err = request.Req(&request.Options{
-			Method: http.MethodPut,
-			Scheme: u.Scheme,
-			Domain: u.Host,
-			Path:   "/sharings/" + s.SID + "/io.cozy.files/" + xoredFileID + "/metadata",
-			Headers: request.Headers{
-				"Accept":        "application/json",
-				"Content-Type":  "application/json",
-				"Authorization": "Bearer " + creds.AccessToken.AccessToken,
-			},
-			Body: bytes.NewReader(body),
-		})
+		res, err = RefreshToken(inst, s, m, creds, opts, body)
```

Comment:
I think that we should close `res.Body` if err is `ErrInternalServerError` or `ErrClientError`